pkg/cli/debug/debug.go

```diff
@@ -595,8 +595,19 @@ func (o *DebugOptions) RunDebug() error {
 			// switch to logging output
 		case err == krun.ErrPodCompleted, err == conditions.ErrContainerTerminated:
 			resultPod, ok := containerRunningEvent.Object.(*corev1.Pod)
```

Comment:
Do you have any solid idea how we can make it smarter;

* not only relying on this casting to gather pod logs(because it might include exceptions like in below)?.
* or this casting is mandatory and `getLogs` would be smart enough for such cases and we'll not have these exceptions below?